wgpu-hal/src/metal/device.rs

```diff
@@ -918,8 +918,20 @@ impl crate::Device<super::Api> for super::Device {
                         self.shared.private_caps.max_vertex_buffers as u64 - 1 - i as u64;
                     let buffer_desc = vertex_descriptor.layouts().object_at(buffer_index).unwrap();
 
-                    buffer_desc.set_stride(vb.array_stride);
-                    buffer_desc.set_step_function(conv::map_step_mode(vb.step_mode));
+                    if vb.array_stride == 0 {
```

Comment:
Could you just add a note about what we're working around here

Reply:
This didn't seem to have happened?

Reply:
Odd, I wrote a comment but it seems when I squashed a few commits I somehow lost this one